From fb618bd27ac7a9fd32973320016c7ff28dd2f60e Mon Sep 17 00:00:00 2001
From: Mike Dalessio <mike.dalessio@gmail.com>
Date: Fri, 5 May 2023 10:10:43 -0400
Subject: [PATCH] rip out libxml2's libc_single_threaded support

This is a preventative measure because this feature relies on a glibc
version we can't realistically require of users today. We're not yet
precompiling on a system with a modern-enough glibc to make this an
actual problem, but I'm doing it while all of this context is fresh.
---
 threads.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/threads.c b/threads.c
index 60dbce4c..eb1c12e5 100644
--- a/threads.c
+++ b/threads.c
@@ -27,7 +27,8 @@
 
 #if defined(HAVE_POSIX_THREADS) && \
     defined(__GLIBC__) && \
-    __GLIBC__ * 100 + __GLIBC_MINOR__ >= 234
+    __GLIBC__ * 100 + __GLIBC_MINOR__ >= 234 && \
+    !defined(NOKOGIRI_PRECOMPILED_LIBRARIES)
 
 /*
  * The modern way available since glibc 2.32.
-- 
2.40.1

